docs/source/notes/autograd.rst

```diff
@@ -1,124 +1,457 @@
 .. _autograd-mechanics:
 
-Autograd mechanics
-==================
+Advanced autograd
+=================
 
-This note will present an overview of how autograd works and records the
-operations. It's not strictly necessary to understand all this, but we recommend
-getting familiar with it, as it will help you write more efficient, cleaner
-programs, and can aid you in debugging.
+This note presents some more advanced topics about the autograd in PyTorch.
+If you are looking for more basic information about autograd, check out
+the `autograd introduction <https://pytorch.org/tutorials/beginner/blitz/autograd_tutorial.html>`_.
 
-.. _excluding-subgraphs:
+.. contents::
+    :local:
+    :depth: 2
 
-Excluding subgraphs from backward
----------------------------------
+Mathematical formulation
+------------------------
```

Comment:
unfortunately, the theme makes it hard to see the difference between the level of captions